### DISCUSSION TITLE
Barcode/ASN scan

### DISCUSSION THREAD

**Discussion by JigSawFr** [LeoKlaus/Paperparrot#171] [Feature Requests]
It can be great to be able to scan barcode from docs to get the related ASN and show the numeric version in paperless.
Thanks !

**Comment by LeoKlaus** [comment]
Are you using Paperless-ngx? If you set up [barcode scanning](https://docs.paperless-ngx.com/configuration/#PAPERLESS_CONSUMER_ENABLE_ASN_BARCODE), ASNs should be recognized.

Paperparrot currently does not recognize barcodes, but it doesn't interfere with the Paperless consumption process.

**Reply by Panzer1119** [reply]
It looks like you've added some kind of this ASN scanning in version 2.1.3, but only for QR codes and data matrices.

Can you extend it to also scan ASN barcodes?

**Reply by LeoKlaus** [reply]
Hey Panzer1119, yeah, sure. I what type of barcode do you use?

**Reply by Panzer1119** [reply]
Personally i'm using `Code 128`.

But if it's not too much work maybe you could implement detecting all [barcode types supported by Paperless-ngx](https://docs.paperless-ngx.com/advanced_usage/#barcodes).

**Reply by JigSawFr** [reply]
Yes please ! I'm also using barcodes

**Reply by LeoKlaus** [reply]
> [barcode types supported by Paperless-ngx](https://docs.paperless-ngx.com/advanced_usage/#barcodes).

I didn't know about that list, that's great! Unless the codes require some form of programmatic decoding, it's literally a single line of code to add support :)

**Comment by LeoKlaus** [comment]
I've support for all barcode types supported by Paperless in a recent update. Closing this now.
